src/sharing/state.js

```diff
@@ -25,10 +43,19 @@ export const revokeSharingLink = permissions => ({
 
 // reducers
 const byIdInitialState = { sharings: [], permissions: [] }
-const updateByIdItem = (state, id, updater) => ({
-  ...state,
-  [id]: updater(state[id] || byIdInitialState)
-})
+const isItemEmpty = item =>
+  item.sharings.length === 0 && item.permissions.length === 0
+const updateByIdItem = (state, id, updater) => {
+  const { [id]: byIdState, ...rest } = state
+  const update = updater(byIdState || byIdInitialState)
+  return isItemEmpty(update)
+    ? rest
+    : {
+        ...rest,
+        [id]: update
+      }
+}
```

Comment:
I get what this function does, but it took me a while. I wonder if just renaming things would be enough, ie.

- updateByIdItem -> updateByItemId
- byIdState -> stateById
- byIdInitialState -> initialItemState

what do you think?

Reply:
I think updateByItemId is not better but worse, we're talking about a `byId` item, not a item ID. So I propose:
 - byIdState -> byIdItemState
 - byIdInitialState -> byIdItemInitialState

Does it look better to you?

Reply:
> I think updateByItemId is not better but worse, we're talking about a byId item, not a item ID

Oooh ok. In that case I think the variables are properly named. And I don't really know how to make it more obvious, so let's go for this.